torch/ao/quantization/backend_config/qnnpack.py

```diff
@@ -11,17 +11,50 @@
     _get_rnn_op_configs,
     _get_share_qparams_op_configs,
 )
-from .backend_config import BackendConfig, DTypeConfig
+from .backend_config import BackendConfig, DTypeConfig, DTypeWithConstraints
 
 
 # ===================
 # |  DTYPE CONFIGS  |
 # ===================
 
-# TODO: For now, these DTypeConfigs are identical to the ones defined in native.py
-# In the future, once we support specifying quant_min/quant_max and scale_min/scale_max,
-# these will diverge. In particular, for QNNPACK, we will restrict the weight quantized
-# values to within [-127, 127] and set the min scale value to 2 ** -12.
+# xnnpack dtype configs
+
+# We restrict scale values to be 2 ** -12 to ensure the
+# requantization scale never falls below the xnnpack lower
+# threshold. Additionally, for qint8 weight, we restrict
+# the quantization values to [-127, +127], excluding -128.
+# For more detail, refer to the description of
+# `default_symmetric_qnnpack_qconfig`.
+
+# TODO: add additional restriction on qscheme to ensure it
+# is either per_tensor_symmetric or per_channel_symmetric
+
+xnnpack_act_qint8_with_constraints = DTypeWithConstraints(
+    dtype=torch.qint8,
+    scale_min_lower_bound=2 ** -12,
+)
+
+xnnpack_weight_qint8_with_constraints = DTypeWithConstraints(
```

Comment:
I dont think we should pollute quantization codebase with backend information anymore than what is already there. So lets not introduce xnnpack here. Can we not just be explicit and name this `weight_qint8_with_symmetric_constraints` and if we want to use backend here maybe say `weight_qint8_with_qnnpack_symmetric_constraints`. Ideally I would like to say `weight_qint8_symmetric_min_scale_2_12` but that is mouthful and confusing.